source/ShopPanel.cpp

```diff
@@ -847,72 +864,75 @@ void ShopPanel::DrawDetailsSidebar()
+	FillShader::Fill(Screen::BottomRight() - .5 * FLEET_BUTTONS_SIZE, FLEET_BUTTONS_SIZE,
 		*GameData::Colors().Get("shop side panel background"));
+	FillShader::Fill(Screen::BottomRight() - Point(SIDEBAR_WIDTH, 0) - .5 * INFO_BUTTONS_SIZE, INFO_BUTTONS_SIZE,
+		*GameData::Colors().Get("shop info panel footer"));
 	FillShader::Fill(
-		Point(Screen::Right() - SIDEBAR_WIDTH / 2, Screen::Bottom() - BUTTON_HEIGHT),
-		Point(SIDEBAR_WIDTH, 1), *GameData::Colors().Get("shop side panel footer"));
+		Point(Screen::Right() - (SIDEBAR_WIDTH + INFOBAR_WIDTH) / 2, Screen::Bottom() - BUTTON_HEIGHT),
+		Point(SIDEBAR_WIDTH + INFOBAR_WIDTH, 1), *GameData::Colors().Get("shop side panel footer"));
 
 	const Font &font = FontSet::Get(14);
 	const Color &bright = *GameData::Colors().Get("bright");
 	const Color &dim = *GameData::Colors().Get("medium");
 	const Color &back = *GameData::Colors().Get("panel background");
 
 	const Point creditsPoint(
-		Screen::Right() - SIDEBAR_WIDTH + 10,
+		Screen::Right() - SIDE_WIDTH + 10,
 		Screen::Bottom() - 65);
 	font.Draw("You have:", creditsPoint, dim);
 
 	const auto credits = Format::CreditString(player.Accounts().Credits());
-	font.Draw({credits, {SIDEBAR_WIDTH - 20, Alignment::RIGHT}}, creditsPoint, bright);
+	font.Draw({credits, {INFOBAR_WIDTH - 20, Alignment::RIGHT}}, creditsPoint, bright);
 
-	const Font &bigFont = FontSet::Get(18);
-	const Color &hover = *GameData::Colors().Get("hover");
-	const Color &active = *GameData::Colors().Get("active");
-	const Color &inactive = *GameData::Colors().Get("inactive");
-
-	const Point buyCenter = Screen::BottomRight() - Point(210, 25);
-	FillShader::Fill(buyCenter, Point(60, 30), back);
-	bool isOwned = IsAlreadyOwned();
-	const Color *buyTextColor;
-	if(!CanBuy(isOwned))
-		buyTextColor = &inactive;
-	else if(hoverButton == (isOwned ? 'i' : 'b'))
-		buyTextColor = &hover;
-	else
-		buyTextColor = &active;
-	string BUY = isOwned ? (playerShip ? "_Install" : "_Cargo") : "_Buy";
-	bigFont.Draw(BUY,
-		buyCenter - .5 * Point(bigFont.Width(BUY), bigFont.Height()),
-		*buyTextColor);
-
-	const Point sellCenter = Screen::BottomRight() - Point(130, 25);
-	FillShader::Fill(sellCenter, Point(60, 30), back);
-	static const string SELL = "_Sell";
-	bigFont.Draw(SELL,
-		sellCenter - .5 * Point(bigFont.Width(SELL), bigFont.Height()),
-		CanSell() ? hoverButton == 's' ? hover : active : inactive);
-
-	const Point leaveCenter = Screen::BottomRight() - Point(45, 25);
-	FillShader::Fill(leaveCenter, Point(70, 30), back);
-	static const string LEAVE = "_Leave";
-	bigFont.Draw(LEAVE,
-		leaveCenter - .5 * Point(bigFont.Width(LEAVE), bigFont.Height()),
-		hoverButton == 'l' ? hover : active);
+	if(playerShips.size() > 1)
+	{
+		const Point textPoint(
+			Screen::Right() - SIDEBAR_WIDTH + 10,
+			Screen::Bottom() - 65);
+		font.Draw(to_string(playerShips.size()) + " ships selected", textPoint, dim);
+	}
+
+	const Interface *shopUi = GameData::Interfaces().Get("shop");
```

Comment:
shopUi isn't a `UI`. It's an `Interface`, so please rename it.

Reply:
99% of other places where Interface::Get() is called, the result is assigned to a variable called fooUi. Why pick on this one to be different?

Reply:
I consider it a design error to have a class called Interface and a second class called UI. I propose renaming UI to PanelStack.